src/registries/npm-registry.js

```diff
@@ -222,13 +222,9 @@ export default class NpmRegistry extends Registry {
   }
 
   getScope(packageIdent: string): string {
-    // Matches '@scope' in a packageName or pathname, otherwise returns ''
-    if (packageIdent && isScopedPackage(packageIdent)) {
-      const matched = packageIdent.match(/(^@|(?!\/)@).+?((?=%2f)|(?=\/))/);
-      return (matched && matched[0]) || '';
-    }
-
-    return '';
+    // Matches strings beginning with @ or contain /@ up until the next forward slash
+    const match = packageIdent.replace('%2f', '/').match(/(^@|\/@)(.+?)(?=\/)/);
```

Comment:
Looking at this closely, I really can't understand a few things:

 - Why do we even have URL-encoded parts in the ident? That doesn't look right so it should be fixed where the value originally comes from.
 - What does the `?` at the end of `(.+?)` part stand for? Is it for the query string delimiter? If so, why do we need to include it in the match?

I feel like the following version is clearer and more accurate:
```
// Matches the first path segment that starts with an `@`. The RegEx is constructed as follows:
// `(?:^|\/)` Match either the start of the string or a `/` but don't capture
// `(@[^?\/]+)` Match a string starting with an `@` and not having any `/` or `?` in it and capture
// `(?:[?\/]|$)/)` Match a path delimiter, a query string delimiter or the end of the string but don't capture
const match = packageIdent.match(/(?:^|\/)(@[^?\/]+)(?:[?\/]|$)/);
return match && match[1] || '';
```